src/nextjournal/clerk/analyzer.clj

```diff
@@ -419,15 +461,54 @@
   Recursively decends into dependency vars as well as given they can be found in the classpath.
   "
   [doc]
-  (let [{:as graph :keys [->analysis-info]} (analyze-doc doc)]
-    (reduce (fn [g [source symbols]]
-              (if (or (nil? source)
-                      (str/ends-with? source ".jar"))
-                (update g :->analysis-info merge (into {} (map (juxt identity (constantly (if source (hash-jar source) {})))) symbols))
-                (analyze-file g source)))
-            graph
-            (group-by find-location (unhashed-deps ->analysis-info)))))
-
+  (loop [{:as state :keys [->analysis-info analyzed-file-set counter]}
+
+         (-> (cond-> doc
+               (not (:graph doc)) analyze-doc)
+             (assoc :analyzed-file-set (cond-> #{} (:file doc) (conj (:file doc))))
+             (assoc :counter 0))]
+    (let [unhashed (unhashed-deps ->analysis-info)
+          loc->syms (apply dissoc
+                           (group-by find-location unhashed)
+                           analyzed-file-set)]
+      (prn :build-graph counter :analyzed-file-set analyzed-file-set)
```

Comment:
There is still a prn here, is that intended?

Reply:
added those to debug windows on CI. Will remove them before merging.